### ISSUE TITLE
Invalid command line arguments are silently ignored (no error message or log message)

### ISSUE THREAD

**Issue by johnbrvc** [pc2ccs/pc2v9#506]
**Describe the issue**: 
If an invalid command line option is supplied to a model at invoke time, it is silently ignored

**To Reproduce**: 
Just supply an invalid command line option to any pc2 command, eg:
pc2admin --bad-option --login a1

**Expected behavior**:
An error message indicating that "--bad-option" is invalid should be displayed and logged and execution should stop.
An invalid option could indicate that the user mistyped something and unexpected results may occur.  Better to let the
user know so it can be corrected. 

**Actual behavior**: 
No message or log message is generated for an invalid option.

**Environment**: please complete the following information:
 All.

**Log Info**:  
No log messages generated on bad options.

**Screenshots**:  
**Additional context**:

**Comment by johnbrvc** [comment]
I looked over Starter.java and ParseArguments.java.  Also, AppConstants.java.  I'm left a bit confused.
It seems that pc2/AppConstants.java has only 5 things it (included here for your convenience):

```
public class AppConstants {
    public static final String HELP_OPTION_STRING = "--help";
    public static final String DEBUG_OPTION_STRING = "--debug";
    public static final String LOGIN_OPTION_STRING = "--login";
    public static final String PASSWORD_OPTION_STRING = "--password";
    public static final String DATETIME_PATTERN_LONG = "E, dd MMM yyyy HH:mm:ss z";
}
```

InternalController.java defines its own identical constant strings for 3 of those (debug, login, password).
It in-lines "--help".  The date pattern is used only by "tools".  It almost looks like at some point, someone
was going to put all the option constant strings in one place (AppConstants - good idea), but it never
materialized.  It seems reasonable (IMHO) to move all the option constants (as a start) from InternalController.java to
AppConstants and use them from there.  Then, a method can be added to AppConstants to verify that
an option is valid, or another class can be created (derived fromAppConstants?) to do the heavy lifting of deciding
if an option is valid. 

Quick hack test attached (rename to .java).  I just did a crude search for all (some regular expression that has "--xxxxx") and dumped everything in the test program - I'm sure there are more options, like the new ones that I have not gotten yet, but
you get the idea.   Optimize: hash table?

[AppConstants.txt](https://github.com/pc2ccs/pc2v9/files/9154441/AppConstants.txt)

**Comment by johnbrvc** [comment]
If no one else wants to look at this issue, I will attempt to fix it in a similar fashion to what I described above, and submit a PR.  Should I assign it to myself or is someone else going to look at it too?

**Comment by johnbrvc** [comment]
I created branch i_506_cli_invalid_opts on my repo to fix this issue.  Also added unit tests to ParseArguments.  Before I make a PR, someone please look over the changes and/or make suggestions.

**Comment by johnbrvc** [comment]
E-Mail received from @lane55 on July 27, 2022 4:09AM Eastern Time:

Overall looks very good.   I have a couple of bumps but once those
are addressed should be ready for a bunch of testing on the 22 classes
that use ParseArguments or a subset thereof.

Attached is an updated Junit because the way it was handlign exceptions
could use improvements to use junit asserts and fails... you'll see.

Major bug in handling the new IllegalArgumentException, program
does not System.exit and when isgui error message is hidden.  I waited
for a while staring an Admin with --bad until I realized that javaw
was gobbling up the message and stack trace.

Requirements are:

1 - When ParseArgument throws the IllegalArgumentException if usingGUI
should show the exception message to the user then System.exit.

2 - When ParseArgument throws the IllegalArgumentException if is not usingGUI
should print message to

3 - In both cases a log message should be written with the exception message.

That.s the bad news, the good news is we already have a method to handle all of
that.  Use a try/catch and fatalError like this, and all 3 requirements are met.

        try {
            parseArguments = new ParseArguments(stringArray, requireArguementArgs, AppConstants.allOptionStrings);
        } catch (Exception e) {
            fatalError(e.getMessage(), e);
        }

re: ParseArguments var name, could be better.

   The dcl is: private String[] validOptions = null;

   These are valid options, but they are more a list of allowed option, right?
Valid options just made me ask what are the validation rules for those options,
rather than know this is a list of the only allowed options.    If the name
is not changed then Javadoc needs to be added to define what 'valid' means.
(or just rename it to allowedOptions or something like that.

re: picky picky

1 - Commit comment incorrect:
Added all command line options to AppConstants class.
should be
Moved all command line options to AppConstants class.

2 - A bit of scope creep, moved other constants into AppConstants.
but a good CI.

   Btw, how would you know, there is already a Constants class
   for package edu.csus.ecs.pc2.core.Constants
   and it was on my list to rename it AppConstants or the like, someday.

Lastly, this change potentially affects all 23 classes that use ParseArguments.
So we should identify which classes other than InternalController will also
be update or add a comment limiting the scope of the PR to jut InternalController.

In either case a number of classes in our documentation should be well tested
(PC2Tools, PC2Report)

When the PR is submitted then I will review every constant that is used
in InternalController and the new constants in AppConstants.

**Comment by johnbrvc** [comment]
Merged into as a result of PR#524.
